File: 2021/01/depth_report.py
def read_input():
    with open('../input/01.txt') as f:
        return [int(line) for line in f.read().split()]

def increases(input):
    return sum(1 for a,b in zip(input, input[1:]) if a < b)

def three_m_sliding_increases(input):
    windows = [sum(input[i:i+3]) for i in range(0, len(input)-2)]
    return increases(windows)

def main():
    input = read_input()
    print(increases(input))
    print(three_m_sliding_increases(input))

if __name__ == "__main__":
    main()

File: 2021/01/depth_report.rb
def increases(input)
  input.each_cons(2).reduce(0) { |m, (a, b)| a < b ? m + 1 : m }
end

def increases_three_m_sliding(input)
  increases input.each_cons(3).map(&:sum)
end

input = File.read('../input/01.txt').split.map(&:to_i)

p increases(input)
p increases_three_m_sliding(input)
